### PR TITLE
Update state to use {country, offset}, hook up chart with redux

### PR DESCRIPTION
**Pull request by vpontis**
- Update state to be `{country: string; offset: number}`
- Add `setRegions` reducer
- Update `Chart.tsx` to read from redux and to use `setRegions`

### CODE REVIEW

File: src/redux/region.tsx [hunk 1]
```diff
@@ -1,35 +1,25 @@
...
+  offset: number;
 }
 
 export function toString(region: IRegion): string {
   let parts: string[] = [region.country];
-  if (region.state !== "") {
-    parts.push(region.state);
-  }
-  parts.push(region.day.format("YYYYMMDD"));
+  parts.push(region.offset.toString());
```
Comment (danqing) [inline src/redux/region.tsx]: if it's only country and offset I guess we don't need to push and then join

File: src/redux/region.tsx [hunk 2]
```diff
@@ -1,35 +1,25 @@
...
-    region.day = dayjs(parts[2], "YYYYMMDD");
-  } else {
-    region.day = dayjs(parts[1], "YYYYMMDD");
-  }
-
-  return region;
+  const [country, _offset] = parts;
+  return {
+    country,
+    offset: parseInt(_offset)
```
Comment (danqing) [inline src/redux/region.tsx]: `parseInt` can return NaN right? shall we default to 0 in such case?

File: src/redux/reducers.tsx [hunk 3]
```diff
@@ -44,27 +46,34 @@ export default function(state: AppState = loadState(), action: TActions) {
...
+    zoom = 1;
+  }
   return persistedState({ ...state, zoom });
 }
 
 function changeMode(state: AppState, to: string): AppState {
   return persistedState({ ...state, mode: to as mode.EMode });
 }
 
+function setRegions(state: AppState, regions: IRegion[]): AppState {
```
Comment (danqing) [inline src/redux/reducers.tsx:67]: I feel like we don't need setRegions (that changes everything at once) - or maybe it's simpler this way? I was thinking addRegion, removeRegion, etc.

Either way, I think for the chart component you don't need to worry about setting the region - the controls should likely be in the x-axis and chart just passively follow the change?

Wdyt?

Comment (vpontis) [reply src/redux/reducers.tsx:67]: Yea, I have some controls in the chart right now. I think those will be replaced by the x-axis that you are making. 

`setRegions` works right now. But feel free to update w/ `addRegion`, `removeRegion` etc as you develop the x-axis

**Pull request merged**